src/samplers/support/hmc_core.jl

```diff
@@ -30,13 +35,35 @@ leapfrog(_θ::Union{Vector,SubArray}, p::Vector{Float64}, τ::Int, ϵ::Float64,
     p_old = p; θ_old = copy(θ); old_logp = getlogp(vi)
 
     p -= ϵ .* grad / 2
-    θ += ϵ .* p  # full step for state
+    θ += ϵ .* p .* (spl.info[:wum][:stds].^2) # full step for state
```

Comment:
@yebai how to use pre-cond (2)

Reply:
there is an error here, the correct way should be 

```
θ += ϵ .* p
```

Reply:
why does Stan do this? I made these changes mainly by referring https://github.com/yebai/Turing.jl/blob/1df604591e4e4297261dfcbb478ca51059ecea9d/src/samplers/support/hmc_core.jl